### ISSUE TITLE
DwC field scientificNameID is not used at all

### ISSUE THREAD

**Issue by bart-v** [gbif/pipelines#217]
**This issue was moved from portal-feedback to pipelines**

Example
https://www.gbif-uat.org/occurrence/search?dataset_key=740cf4e0-37ca-4389-ba8f-4e1bc5177893&taxon_key=5401803

Lists the records as Oligochaeta and appends the authority "K.Koch" just like that.
That makes these marine occurrences terrestrial plants... 

While a scientificNameID urn:lsid:marinespecies.org:taxname:2036 is provided, that can be resolved to the animal class Oligochaeta.

This is a missed chance to fix homonyms in an easy way...

**Comment by timrobertson100** [comment]
For current links, all [Danish Mycological Society, fungal records database](https://www.gbif.org/dataset/84d26682-f762-11e1-a439-00145eb45e9a) records contain `scientificNameID` pointing to [Index Fungorum](http://www.indexfungorum.org/Names/NamesRecord.asp?RecordID=327981) such as [this example](https://www.gbif.org/occurrence/2237852364).

**Comment by bart-v** [comment]
@mdoering about finding out what checklist (version) has been used, everything is solved by using a proper and persistent GUID (like LSID): it tells you what authority has been used, on a per record basis.

I don't understand this question
> if we rely on globally unique ids (...) how do we know which checklist is the authority in case several checklists use these ids?

If it's a GUID, there is only one single checklist who has assigned/generated this GUID, so there is nothing to choose from?

**Comment by bart-v** [comment]
Thanks @timrobertson100

**Comment by mdoering** [comment]
@bart-v a properly versioned LSID would tell you what it was when resolving it. But I doubt a DwC WoRMS archive contains all historical versions of a name or deleted names. 

My point about a non unique GUID is that there might be various datasets, e.g. molluscabase, WoRMS, Catalogue Of Life that all use the same GUID. Knowing which is the authorative one seems trivial by looking at the domain, but I would expect we better have some metadata about that on the dataset level. I am sure GUIDs will not appear once only.

**Comment by bart-v** [comment]
WoRMS could do versions but that is usually overkill. We hardly ever change names, but create new ones ans point to them to each other. We do keep track of deletions.

I agree that some metadata on dataset level is needed, indeed.

**Comment by auspex** [comment]
> My point about a non unique GUID is that there might be various datasets, e.g. molluscabase, WoRMS, Catalogue Of Life that all use the same GUID. Knowing which is the authorative one seems trivial by looking at the domain, but I would expect we better have some metadata about that on the dataset level. I am sure GUIDs will not appear once only.

There can't be a "non unique GUID". It's in the name: "Globally unique..."
As Bart says at the top, his taxa are **urn:lsid:marinespecies.org:...** and those in Tim's example are **urn:lsid:indexfungorum:...** 

I don't think it matters which name list is authorative! Only that the user can see which was used. As they can, when the **urn:lsid:** format is used. _[Note: To be fair, **our** `scientificNameID` are in the form https://www.marinespecies.org/aphia.php?p=taxdetails&id=1, as required by EurOBIS, which we have argued is wrong, particularly when they use **urn:** for other vocabs!]_

The most distressing thing about this issue is that i can see the simple solution to my #934 is to **remove** `scientificName` from my datasets! It will make the data less useful to GBIF but at least it won't be wrong! And OBIS will be happy. 

In an case, it's wrong for GBIF to make assumptions abut my data.

**Comment by timrobertson100** [comment]
Hi folks 

To try and address some of the challenges I think we could make a good step forward with a fairly simple solution. What do people think about the following, please?

Taking [this record](https://www.gbif.org/occurrence/4059201575) as an example, it comes with:
```
scientificName: Megaptera novaeangliae
scientificNameID: urn:lsid:marinespecies.org:taxname:137092
```

In the processing we could do the following:

1. Detect that `scientificNameID` contains an identifier we've enabled in configuration based on the prefix of `urn:lsid:marinespecies.org`
2. We'd look that up against the reference checklist (we'd configure that prefix to point to the WoRMs checklist) using [this API call](https://api.gbif.org/v1/species?datasetKey=2d59e5db-57ad-41ff-97d6-11f5fb264527&sourceId=urn:lsid:marinespecies.org:taxname:137092)
3. The response has the `nubKey` (the backbone key) which we'd then use to populate the names and necessary backbone identifiers for the record

This approach would use the identifier mapping to find things in the GBIF backbone which is a more robust mapping than the names-based lookup service.

There will always be some inconsistency due to the publishing cycle (e.g. occurrence records with names not in the latest WoRMS dataset) but it would at least 1) improve the homonym cases, and 2) improve the cases where only IDs are provided.

To get a sense of which prefixes would be suitable to map against a checklist please see this:
```
SELECT substring(scientificNameID, 1, 15) as prefix, count(*) AS records 
FROM prod_h.occurrence 
GROUP BY substring(scientificNameID, 1, 15) 
HAVING count(*)>250000 
ORDER BY records DESC

(removing some noise) yields:

urn:lsid:marine	66526389
urn:lsid:itis.g	15637610
urn:lsid:dyntax	2303347
urn:lsid:biosci	1454750
urn:lsid:indexf	1065547
urn:lsid:ipni.o	448138
http://www.mari	296647
```

What do you think? Thanks

**Comment by derek-mba** [comment]
That looks good to me. That last row returned by your query is probably all from datasets submitted to EurOBIS!

> There will always be some inconsistency due to the publishing cycle (e.g. occurrence records with names not in the latest WoRMS dataset)

I had to find the ID in WoRMS before I published the dataset. The only time that could happen is if the record was deleted, but that would be exceedingly rare (generally, invalid taxa are _flagged_ as invalid but ironically an invalid ID is just as valid for the purpose!). I would expect other authorities to do the same.

**Comment by bart-v** [comment]
Perfect @timrobertson100  !

**Comment by ymgan** [comment]
This is great @timrobertson100 !! Thank you so much for this! I appreciate it!

**Comment by derek-mba** [comment]
@rubenpp7 You're right, I got that backwards. In which case, I can't help reduce the inconsistency: ours are `urn:lsid:marinespecies.org:taxname:`

**Comment by rubenpp7** [comment]
@timrobertson100 thanks a lot for the list,
I have randomly checked 12 of those datasets and all of them are Published by the [Australian Antarctic Data Centre](https://www.gbif.org/publisher/3693ff90-4c16-11d8-b290-b8a03c50a862). I'll point them at this github issue 😄

**Comment by timrobertson100** [comment]
Thanks. To save effort - two publishers are involved. The other is [this one](https://www.gbif.org/pt/publisher/1c9c3bb4-377c-4547-aa79-16ea755d6006) with 3 records.

> I'll point them at this github issue

Thank you

**Comment by rubenpp7** [comment]
Great, I contacted them as well, hopefully we will get some answers from themselves here soon.

Cheers

**Comment by timrobertson100** [comment]
I've put together an implementation and have started running some tests on data. Please can I have some guidance on how people would expect this to work for this kind of situation?

Given a record with the following:

```
scientificNameID: urn:lsid:marinespecies.org:taxname:1328690
kingdom: Animalia
phylum: Mollusca
class: Gastropoda Cuvier, 1797
order: Stylommatophora Schmidt, 1856
family: Helicidae Rafinesque, 1815
genus: Helix Linnaeus, 1758
scientificName: Helix paromphala Lowe,
```

In production _today_, we ignore `scientificNameID` and run [this lookup](https://api.gbif.org/v1/species/match2?kingdom=Animalia&phylum=Mollusca&class=Gastropoda%20Cuvier,%201797&order=Stylommatophora%20Schmidt,%201856&family=Helicidae%20Rafinesque,%201815&genus=Helix%20Linnaeus,%201758&scientificName=Helix%20paromphala%20Lowe,) which detects it to be [_Helix poromphala_ R.T.Lowe, 1852](https://api.gbif.org/v1/species/11314464), a synonym of [_Discula polymorpha poromphala_ (R.T.Lowe, 1852)](https://api.gbif.org/v1/species/9542548). Assuming _p**a**romphala_ is a misspelling(???) of _p**o**romphala_ the name-based approach seems to agree with [WoRMS](https://www.marinespecies.org/aphia.php?p=taxdetails&id=1504094).

However, if we use the `scientificNameID` which is the WorMS concept of [_Euhadra peliomphala_ (L. Pfeiffer, 1850)](https://www.marinespecies.org/aphia.php?p=taxdetails&id=1328690) the [lookup](https://api.gbif.org/v1/species?datasetKey=2d59e5db-57ad-41ff-97d6-11f5fb264527&sourceId=urn:lsid:marinespecies.org:taxname:1328690) would lead us to the [_Euhadra peliomphala_ (L.Pfeiffer, 1850)](https://api.gbif.org/v1/species/8977527) concept in GBIF (note it has the basionym of _Helix peliomphala_ L.Pfeiffer, 1850 which might suggest some fuzzy name match or perhaps mistyping when the person was looking up the ID to put on the record?).

This is one of several examples in the first records I've explored that seem to have contradictions between the name and what the ID resolves to. Please do point out anything I am doing obviously wrong too and bear in mind all these names are unknown to me.

Thanks

**Comment by auspex** [comment]
I'd say, there's likely something wrong with that, in any case, as the first thing that happens when I go to the WoRMS link is that it tells me _"Oops! This taxon is out of scope! The taxon you have searched for is non-marine."_ Which suggests it's either not what it was idenitified as, or WoRMS is not the definitive source for identification (in this case, _poromphala_ is also terrestrial, so perhaps the urn should have been MolluscaBase rather than WoRMS). 

Personally, I don't think GBIF should even attempt to fill in the hierarchy if the `scientificName` and `scientificNameId` don't match. In this case, the `scientificName` doesn't even directly match anything in either WoRMS or GBIF. If GBIF _does_ add the hierarchy, or finds any instance of fields not matching what you'd expect, I'd appreciate an email with a summary of the fields added or issues found.

I feel your approach is right--in this case it's simply identifying a case where there needs to be more QA of the dataset.

**Comment by derek-mba** [comment]
> I'd say, there's likely something wrong with that,
Oops. That's me again... I keep forgetting to use the correct GitHub account.

**Comment by albenson-usgs** [comment]
Also yes this is likely a fuzzy match issue. When I put "Helix paromphala" into the quick search the first result it gives me is the one the scientificNameID is identifying.

![Capture](https://github.com/gbif/pipelines/assets/7586150/d768c0c9-3d00-4226-aef9-2f31aae1fe6c)

**Comment by derek-mba** [comment]
Yes, obviously WoRMS _does_ have terrestrial species, but my point is that if the contributor is providing terrestrial data, WoRMS was likely not the right source for looking up the taxonomy. When I populate a dataset, that's one of the first QA checks: "is it marine?"

**Comment by bart-v** [comment]
For the example of Tim: 

**Notice**: I'm assuming that scientificNameID is coming from an occurrence record
We don't know if the name is coming from a OBIS dataset or not, so we cannot judge on marine vs. non-marine. And yes WoRMS (Aphia) contains non-marine taxa too.

The process in the example is just fine:
1. give priority to the scientificNameID, and map it to urn:lsid:marinespecies.org:taxname:1328690 `Euhadra peliomphala (L. Pfeiffer, 1850)`, since somebody put effort to standardize it.
2. mark the record as "an issue"
3. later updates might update or correct the LSID

This is actually a bad example (or user error), because the correct LSID is here
https://www.marinespecies.org/aphia.php?p=taxdetails&id=1504094 `urn:lsid:marinespecies.org:taxname:1504094`

In both approaches (using ID vs name lookup) the outcome is currently not correct.

So maybe another example @timrobertson100 ? :)

**Comment by timrobertson100** [comment]
Thanks @bart-v, all

> In both approaches (using ID vs name lookup) the outcome is currently not correct.

I'm not sure about this. I might overlook something, but the name-based approach seemingly does make that match (from above):

> ... run [this lookup](https://api.gbif.org/v1/species/match2?kingdom=Animalia&phylum=Mollusca&class=Gastropoda%20Cuvier,%201797&order=Stylommatophora%20Schmidt,%201856&family=Helicidae%20Rafinesque,%201815&genus=Helix%20Linnaeus,%201758&scientificName=Helix%20paromphala%20Lowe,) which detects it to be [Helix poromphala R.T.Lowe, 1852](https://api.gbif.org/v1/species/11314464), a synonym of [Discula polymorpha poromphala (R.T.Lowe, 1852)](https://api.gbif.org/v1/species/9542548).

Regardless of that, @Markus and I agree with what you outline. I suspect I will have questions about how to apply an issue - e.g. I anticipate we'll see many `scientificName` containing abbreviated authorship when compared to the WoRMS concept name. Do we flag that? If not, we might find we get false positives/negatives in the flagging routine.

> So maybe another example @timrobertson100 ? :)

I'll generate a report of everything that would change (without flagging, but that will come). It might serve as a useful report to approach the relevant publishers which would improve DQ in both GBIF and OBIS.

**Comment by bart-v** [comment]
Sorry Tim, you are correct the name lookup is OK.
Still that is only one example...

**Comment by timrobertson100** [comment]
Using [this utility](https://github.com/gbif/key-value-store/blob/scientificNameID/kvs-gbif/src/test/java/org/gbif/kvs/species/NameIdImpactUtil.java) I looked up the `293,862` unique classifications having WoRMS LSID. This [result](https://github.com/gbif/pipelines/files/12368501/worms-changed.txt) lists the `19,042` that would change the resulting taxon the record is linked to - I haven't reviewed this yet myself beyond a cursory scan and check of a few records.

The file contains:
- the name id and classification seen on the occurrence records
- the number of records having that classification
- the current name and accepted name we match to (i.e. ignoring the name ID)
- the new and accepted names we'd match if the change were to go live

We're going to have to determine if - on balance - this seems like a good set of changes bearing in mind there will always be outliers and misuse.

This does not yet include any flagging,

**Comment by Mesibov** [comment]
@timrobertson100, many thanks for pulling those out and demonstrating the need for a disagreements protocol (https://discourse.gbif.org/t/millipedes-in-the-ocean/3991/7). Some of those scientificName entries have 15+ different scientificNameIDs from WoRMS.

The 19042 records are also very messy in the higher taxon fields and have numerous disagreements there, so I hope GBIF isn't planning to use those higher-taxon fields for resolving disagreements.

FYI, in data auditing DwC datasets for Pensoft data papers, nearly every time there's a scientificNameID field there are disagreements - either multiple sNIDs for sNs, or simply an incorrect sNID choice.

**Comment by timrobertson100** [comment]
Thanks for taking the time to explore @Mesibov - those are helpful insights from your auditing experience.

To add - one disagreement I've spotted seems to be where the scientificNameID points to an accepted concept, where then names are using a synonym. Another common one seems to be similar names (misspellings perhaps ?) that lead to a bad ID lookup as we _suspect_ from the [_Helix paromphala_ example above](https://github.com/gbif/pipelines/issues/217#issuecomment-1680684346)

**Comment by mdoering** [comment]
Can someone shed light on how WoRMS identifiers are assigned to OBIS records? Is this a purely manual process or are scripts, fuzzy lookups etc involved?

**Comment by ymgan** [comment]
Thank you so much for looking into this! I don't want to say anything that I am unsure of, but @pieterprovoost should be the best person to answer this when he is back next week.

**Comment by bart-v** [comment]
WoRMS identifiers in OBIS can be assigned in different ways
Ordered by most->least common
1. taxon match https://www.marinespecies.org/aphia.php?p=match
2. using the API directly https://www.marinespecies.org/rest/, or via clients packages like https://docs.ropensci.org/worrms/ consuming the API
3. manual

Fuzzy lookups are supported in (1) & (2)

**Comment by MattBlissett** [comment]
I think we should interpret using the scientificNameId, as it's then in the publisher's power to change their data unambiguously — and generally, if we receive identifiers we should expect them to have been assigned carefully.

An issue (TAXON_MISMATCH like COUNTRY_MISMATCH? Or TAXON_IDENTIFIER_CONFLICT?) where the lookup based on scientificNameId conflicts with that made on the name string parts is useful.

**Comment by mdoering** [comment]
100% agree with @MattBlissett

**Comment by Mesibov** [comment]
@MattBlissett: "An issue (TAXON_MISMATCH like COUNTRY_MISMATCH? Or TAXON_IDENTIFIER_CONFLICT?) where the lookup based on scientificNameId conflicts with that made on the name string parts is useful."

As demonstrated earlier, this is _necessary_, not just useful, because although you might "expect them [IDs] to have been assigned carefully", @timrobertson100 has provided data to show that in practice the coupling of name to ID can be sloppy.

For this flagging of an issue, are you also proposing that GBIF track _all_ the scientificNameID sources and compare the one selected for a record with the name, or just WoRMS APHIA IDs?

Also, what will happen to the original scientificName and scientificNameID entries in a record where GBIF detects a conflict? Will both continue to appear in the interpreted record?

**Comment by Mesibov** [comment]
Further: @timrobertson100 found records with defective WoRMs IDs, like "urn:lsid:marinespecies.org:taxname:0000000000000000614620". Would that be another, separate issue to flag?

**Comment by mdoering** [comment]
> Further: @timrobertson100 found records with defective WoRMs IDs, like "urn:lsid:marinespecies.org:taxname:0000000000000000614620". Would that be another, separate issue to flag?

Yes!

**Comment by ManonGros** [comment]
I agree that we should have a "scientific name and identifier mismatch". In case of conflict, it would probably make sense to privilege the scientific name over the ID because it would make it more transparent to users. A scientific name is human readable and doesn't require checking an external source.

--

Not sure about the example above ("urn:lsid:marinespecies.org:taxname:0000000000000000614620") where it looks like we have a WoRM prefix but the identifier doesn't exist. Is it something we could check and flag?

Either, we could have a a general warning for "scientific name ID not matched". But in that case, any scientific name ID that aren't in the sources we check would be flagged.
Or we could have a flag only for the scientific names IDs where we have a prefix that we recognise which might be a bit difficult to implement?

**Comment by mdoering** [comment]
> I agree that we should have a "scientific name and identifier mismatch". In case of conflict, it would probably make sense to privilege the scientific name over the ID because it would make it more transparent to users. A scientific name is human readable and doesn't require checking an external source.

I would argue the other way around and prefer the identifier, as this is less of an interpretation from the GBIF side and entirely in the hands of the publisher.

**Comment by Mesibov** [comment]
@MattBlissett , please look at https://www.gbif.org/occurrence/4399021367 (from a dataset I happened to be checking today).

The publisher gives "Sterna anaethetus" and ID "urn:lsid:marinespecies.org:taxname:212605".
The accepted name is "Onychoprion anaethetus" with ID "urn:lsid:marinespecies.org:taxname:567792".
GBIF has interpreted (added) the genus as "Onychoprion" and interpreted (added) the taxonomicStatus as "Synonym", but has not changed either the original scientificName or the original scientificNameID.

(1) How would this record look if GBIF preferenced ID over name?
(2) What would happen if the publisher had "corrected" the ID but not the name, i.e. gave "Sterna anaethetus" and "urn:lsid:marinespecies.org:taxname:567792"?

**Comment by timrobertson100** [comment]
Thank you all for guiding this work. 

> are you also proposing that GBIF track all the scientificNameID sources and compare the one selected for a record with the name, or just WoRMS APHIA IDs?

Not quite. I suggest we use configuration to enable certain patterns e.g. `urn:lsid:marinespecies.org:*` mapped to WoRMS. I think it would be wise to do an impact assessment like we did above, before enabling others. I anticipate IPNI, DynTaxa, IndexHerbariorum would be good candidates though. 

> what will happen to the original scientificName and scientificNameID entries in a record where GBIF detects a conflict? Will both continue to appear in the interpreted record?

Like @mdoering I think I would lean towards the values found using the `scientificNameID` since it's less ambiguous and puts the responsibility and control on the publisher - especially if we provide sensible flags that make it easy to detect and fix. The verbatim values are always available of course.

Taking into consideration everyone's comments, I think we would need the following flags to be transparent and to make it easy to locate problematic records. 

| **Issue**                      | **Description**                                                                                                                                                                                                                                                                                                                                                                             |
|--------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `SCIENTIFIC_NAME_ID_IGNORED` | The `scientificNameID` uses a pattern that is not configured in GBIF. The backbone lookup was performed using the names on the record.  | 
| `TAXON_CONCEPT_ID_IGNORED` | The `taxonConceptID` uses a pattern that is not configured in GBIF. The backbone lookup was performed using the names on the record.  | 
| `SCIENTIFIC_NAME_ID_NOT_FOUND` | The `scientificNameID` matched a known pattern, but it was not found in the associated checklist. The backbone lookup was performed using the names on the record ignoring the ID. This may indicate a poorly formatted identifier or may be caused by a newly created ID that isn't yet known in the version of the published checklist.                                                                                                                                    |
| `TAXON_CONCEPT_ID_NOT_FOUND`   | The `taxonConceptID` matched a known pattern, but it was not found in the associated checklist. The backbone lookup was performed using the names on the record ignoring the ID. This may indicate a poorly formatted identifier or may be caused by a newly created ID that isn't yet known in the version of the published checklist.                                                                                                                                     |
| `SCIENTIFIC_NAME_CONFLICT`     | The `scientificName` on the occurrence record does not exactly match the `scientificName` found in the registered checklist when located using the declared ID (`scientificNameID` or `taxonConceptID`). Publishers are advised to check the ID is correct, or update the formatting of the names on their records.                                                                         |
| `TAXON_ID_CONFLICT`            | The GBIF Backbone concept was found using the `scientificNameID` or `taxonConceptID` but it differs from what would have been found if the classification names on the record were used. This may indicate a gap in the GBIF backbone, a poor mapping between the checklist and the backbone, or a mismatch between the classification names and the declared IDs (`scientificNameID`  or `taxonConceptID`).  | 

Please keep the feedback coming - especially if you disagree. Thank you.

**Comment by Mesibov** [comment]
@timrobertson100, so the answer to my first question is "No different"?

**Comment by mdoering** [comment]
We would lookup the WoRMS ID [212605](https://marinespecies.org/aphia.php?p=taxdetails&id=212605) in the checklist published to GBIF: https://www.gbif.org/species/155305680/verbatim

This checklist (like all others) is matched to the backbone and we can take that mapping to find the matching backbone entry for that _name_, which would be the `nubKey` property in this API response: https://api.gbif.org/v1/species/155305680

In that case we miss that matching for unknown reasons. I will investigate, because the backbone does have the name [Sterna anaethetus Scopoli, 1786](https://www.gbif.org/species/5229256) listed sourced from ITIS and regarded as a synonym of [Onychoprion anaethetus subsp. anaethetus (Scopoli, 1786)](https://www.gbif.org/species/7192428).

The point is that we will only link via name and use the name as it is classified in the backbone, not in the source.

If the publisher would move the identifier to the accepted name id, the occurrence would be listed as _Onychoprion anaethetus_ instead and if the given name would remain as _Sterna anaethetus_, a mismatch flag would be risen.

**Comment by MattBlissett** [comment]
> The point is that we will only link via name and use the name as it is classified in the backbone, not in the source.

Or in other words, the "Interpreted" name is the name according to the GBIF backbone, whether we found it by matching strings or looking up an identifier.

Bob's other point:

> GBIF has interpreted (added) the genus as "Onychoprion" and interpreted (added) the taxonomicStatus as "Synonym", but has not changed either the original scientificName or the original scientificNameID.

Should we be blanking the interpreted scientificNameID in the situations where it is not used? i.e. for `SCIENTIFIC_NAME_ID_IGNORED` and `SCIENTIFIC_NAME_ID_NOT_FOUND`? It remains, of course, in the "Original"/verbatim data.

**Comment by Mesibov** [comment]
@timrobertson100 and @mdoering , leaving aside the interesting question of why WoRMS should get priority in this proposed change, please note that the OBIS manual (https://manual.obis.org/darwin_core.html#taxonomy-and-identification) says

"scientificName (required term) should always contain the originally recorded scientific name, even if the name is currently a synomym [sic]. This is necessary to be able to track back records to the original dataset... A [WoRMS](http://www.marinespecies.org/) LSID should be added in scientificNameID (required term), OBIS will use this identifier to pull the taxonomic information from the World Register of Marine Species (WoRMS) into OBIS, such as the taxonomic classification and the accepted name in case of invalid names or synonyms."

I read that as saying that OBIS contributors can use the original scientificName and the accepted scientificNameID, which guarantees mismatches. Maybe @albenson-usgs could comment?

**Comment by Mesibov** [comment]
@mdoering, OK, so an OBIS user enters the original name and the LSID for that original name, and if it's a synonym, then OBIS replaces the original name and ID with the accepted name and ID? Which of these goes into the OBIS records shared with GBIF - the original name + its ID, or the accepted name +its ID?

**Comment by ymgan** [comment]
> I read that as saying that OBIS contributors can use the original scientificName and the accepted scientificNameID, which guarantees mismatches.

Thank you Bob! OBIS nodes use the scientificNameID of the original scientificName. For example:

scientificName: Diogenichthye 
scientificNameID: [urn:lsid:marinespecies.org:taxname:397410](https://www.marinespecies.org/aphia.php?p=taxdetails&id=397410)

We were specifically asked to **not** change the scientificName to its accepted name or/and uses the lsid of the accepted name as shown below:

scientificName: Diogenichthys 
scientificNameID: [rn:lsid:marinespecies.org:taxname:125820](https://www.marinespecies.org/aphia.php?p=taxdetails&id=125820)

**Comment by Mesibov** [comment]
@mdoering, does the GBIF backbone perfectly follow the checklists (allowing for update delays) in all cases, so that matching is done with names and classifications in checklists, and the checklists amount to subsets of the backbone?

**Comment by Mesibov** [comment]
@ymgan, which of these goes into the OBIS records shared with GBIF - the original name + its ID, or the accepted name +its ID?

**Comment by ymgan** [comment]
> which of these goes into the OBIS records shared with GBIF - the original name + its ID, or the accepted name +its ID?

The former goes to GBIF and OBIS: original name + its ID

scientificName: Diogenichthye
scientificNameID: [urn:lsid:marinespecies.org:taxname:397410](https://www.marinespecies.org/aphia.php?p=taxdetails&id=397410)

**Comment by mdoering** [comment]
> @mdoering, does the GBIF backbone perfectly follow the checklists (allowing for update delays) in all cases, so that matching is done with names and classifications in checklists, and the checklists amount to subsets of the backbone?

The backbone is currently only updated twice per year using most of the important lists, but not everything there is out there. WoRMS, ITIS, IPNI & ZooBank are included for example. That means there clearly are names not included in the backbone, especially all newly described ones.

The lists that are candidate for name identifiers are updated in various frequencies defined by their publishers. Whenever a new version of a list is imported we match it to the backbone during the import process.

**Comment by Mesibov** [comment]
Thank you, @ymgan. Does OBIS share with GBIF like this?

scientificName = original name
scientificNameID = original name's ID
acceptedNameUsage = accepted name
acceptedNameUsageID = accepted name's ID

That seems very clear to me. I'm finding the proposed changes to GBIF's interpretation protocol confusing, unless GBIF would do

scientificName = interpreted name
scientificNameID = interpreted name's ID
verbatimScientificName (not DwC) = original name
verbatimScientificNameID (also not DwC) = original name's ID

**Comment by ymgan** [comment]
Thank you @Mesibov for these interesting questions!

> Does OBIS share with GBIF like this?
> 
> scientificName = original name
> scientificNameID = original name's ID
> acceptedNameUsage = accepted name
> acceptedNameUsageID = accepted name's ID

According to my understanding, no - because the accepted name and accepted name's ID may become unaccepted again in the future. So we are trained/asked to provide scientificName, scientificNameID along with scientificNameAuthorship, kingdom, taxonRank, taxonRemarks of the original name.

**Comment by Mesibov** [comment]
Thank you, @ymgan. So the looking-up of the original name's ID in WoRMS is done twice, independently. OBIS does it (according to the Manual) for its own purposes but doesn't share the result. GBIF does it and may change what it receives from OBIS (in the case of unaccepted names) but does not make clear what's happened (see the "Sterna anaethetus" example, above).

**Comment by timrobertson100** [comment]
> and may change what it receives from OBIS

It may just be the phrasing, but there may be a misunderstanding to clarify. A data publisher publishes a dataset, which is registered in the GBIF registry and linked to the [OBIS network](https://registry.gbif.org/network/2b7c7b4f-4d4f-40d3-94de-c28b6fa054a6). Both GBIF and OBIS ingest the same dataset from the source (generally a GBIF IPT) and process it for the services each infrastructure provides.

**Comment by Mesibov** [comment]
@timrobertson100, thank you for that clarification. I misinterpreted the source of the record I cited above. The publisher (https://www.gbif.org/publisher/5fa89f68-9af0-4a0d-8998-ea39695c1db9) is "CSIRO NCMI IDC / OBIS Australia", so I assumed that "OBIS Australia", a node in the OBIS network, was co-publisher.

**Comment by Mesibov** [comment]
OBIS provides useful information on the fields it provides for its own downloads:  https://obis.org/data/access/.

scientificName is derived from the provided scientificNameID, but there is also an originalScientificName field for the name as provided. There is an APHIA ID field with "the valid name based on the scientificNameID or derived by matching the provided scientificName with WoRMS".

I'm not sure I follow the bit after "or". I think it might apply in cases where (contrary to the requirement) only a name was provided and not an ID. There is no originalScientificNameID field.

OBIS also checks name and ID fields for quality: https://github.com/iobis/obis-qc. 
OBIS checks
- to ensure that scientificNameID is a valid WoRMS ID
- to ensure that scientificName and scientificNameID match in WoRMS

So yes, the same tests would be done twice, independently, for datasets published through OBIS and through GBIF, if GBIF goes ahead with preferencing ID to name in its lookups.

**Comment by ManonGros** [comment]
I think the proposed flags https://github.com/gbif/pipelines/issues/217#issuecomment-1696930580 are sensible.
I can see the point of prioritising the IDs for interpretation (in the cases where they don't match the name). If we get too many confused users, we should consider changing the behaviour or rolling back on the changes.

**Comment by ymgan** [comment]
Thank you so much!! I agree with @ManonGros 

I think the proposed flags https://github.com/gbif/pipelines/issues/217#issuecomment-1696930580 are sensible too. Maybe it would come clear when we see how these flags apply to the examples encountered.

Just throwing out idea, I am wondering if it make sense for the flags can follow the vocab from BDQ TG2 https://github.com/tdwg/bdq/issues/152#issue-354943638 ? For example:

- `SCIENTIFIC_NAME_AND_ID_INCONSISTENT`
- `TAXON_MATCH_NAME_AND_ID_AMBIGUOUS`

Do these capture the same meaning? 

On the other hand, if there are things worth to be mentioned in OBIS manual that could help to *prevent* certain issues identified in the test run, it will be great if someone could create an issue at https://github.com/iobis/manual (I hope this doesn't side track the conversation)

Thank you so much again!

**Comment by timrobertson100** [comment]
Thanks @ymgan - That makes good sense. I'll adjust the labels above accordingly.

**Comment by Mesibov** [comment]
@albenson-usgs, many thanks for your comments.

Please note that verbatimIdentification is not the same as "verbatimScientificName". vI (https://dwc.tdwg.org/terms/#dwc:verbatimIdentification) is a place for informal names, guesses, vernacular names etc as well as formal scientific names and "is meant to be used **in addition to** dwc:scientificName (and dwc:identificationQualifier etc.), not instead of it".

vI is a handy field for data checkers because it allows us to say to compilers: "coral sp. 1a is not appropriate for scientificName. Please put coral sp. 1a in verbatimIdentification and put a formal scientific name for the coral taxon in scientificName".

A record could therefore have
- an original ID (in verbatimIdentification)
- an original scientificName
- an original scientificNameID
- an interpreted scientificName
- an interpreted scientificNameId

I suggest that to avoid confusion, all 5 fields should be present in the record made available to data end-users.

**Comment by timrobertson100** [comment]
Thank you all for contributing so actively to this thread.

I'll now implement those flags, prepare a configuration that handles the WoRMS LSIDs, and process all datasets using them. I am sure we'll refine this again, but getting GBIF and OBIS more closely aligned should be a good start.

I have a remaining question - How strictly should the `scientificName` comparison be when flagging differences?

Consider e.g. a record with `Aus bus` in the name, and an LSID that returns `Aus bus L. 1771`.  Should I flag anything not exactly the same, or should I [parse both sides](https://www.gbif.org/tools/name-parser) and compare the resulting canonical? I'm tempted to suggest the latter as a start to avoid too many nuisance flags, perhaps making it stricter in the future. Thoughts?

**Comment by Mesibov** [comment]
@timrobertson100, please report back here or in the GBIF forum/data blog with the flag tallies you find for records with WoRMS IDs.

DwC scientificName's recommendation is "with authorship and date information if known". Authorship and authorship/date are missing in a large proportion of datasets I see or are put in scientificNameAuthorship. Please parse to canonical. I'm assuming your parser will deal properly with "subsp/subsp./ssp/ssp...." etc.

**Comment by timrobertson100** [comment]
Thanks, @Mesibov - that was my intuition too.

**Comment by mdoering** [comment]
My understanding of `TAXON_MATCH_NAME_AND_ID_AMBIGUOUS` is that this compares the backbone usage _key_, i.e. integer, not name strings. Correct?

And `SCIENTIFIC_NAME_AND_ID_INCONSISTENT` would compare the plain name & author string, not its classification, of the occurrence vs the WoRMS name (or whatever other source was used for the name identifier). But _not_ with the name as interpreted using the GBIF backbone. Is that correct? So the interpreted name (according to the backbone) can still be slightly different to the dwc:scientificName of the OBIS occurrence and SCIENTIFIC_NAME_AND_ID_INCONSISTENT would _not_ be flagged - if WoRMS uses the same scientificName as the occurrence. Is this what we want? Or do we want to flag occurrences whos _interpreted_ scientificName using the backbone differs?

**Comment by Mesibov** [comment]
@mdoering, don't you think that comparing _interpreted_ values will risk piling one error on top of another? Interpreted values are sometimes wrong, and in any case are not the responsibility of the data publisher.

**Comment by pieterprovoost** [comment]
Sorry, I have been away for a while.

> So the looking-up of the original name's ID in WoRMS is done twice, independently. OBIS does it (according to the Manual) for its own purposes but doesn't share the result.

The results are shared. OBIS currently performs lookup of the provided `scientificNameID` in WoRMS (usually a WoRMS LSID but can also be a BOLD or NCBI ID), or matches the scientific name using the WoRMS API in case the ID is missing or invalid. We then replace the full taxonomy with the taxonomy of the accepted name. So the user sees:

- `AphiaID`: interpreted WoRMS ID for the accepted name
- `scientificName` and higher ranks: accepted names according to WoRMS
- `originalScientificName`: `scientificName` as provided
- `scientificNameID`: `scientificNameID` as provided

For consistency we should probably rename `AphiaID` to `scientificNameID` and `scientificNameID` to `originalScientificNameID` (or some other form indicating verbatim value).

> I read that as saying that OBIS contributors can use the original scientificName and the accepted scientificNameID, which guarantees mismatches. 

No, if the name is a synonym we recommend providing the ID for the synonym.

> Consider e.g. a record with Aus bus in the name, and an LSID that returns Aus bus L. 1771. Should I flag anything not exactly the same, or should I parse both sides and compare the resulting canonical? I'm tempted to suggest the latter as a start to avoid too many nuisance flags, perhaps making it stricter in the future. Thoughts?

Yes, please use the canonical. The OBIS recommendation is still to not provided the authorship (despite the DwC definition). 

> https://github.com/gbif/pipelines/issues/217#issuecomment-1696930580

The flags make sense, I'll see if we can implement them as well.

**Comment by Mesibov** [comment]
@pieterprovoost, many thanks for participating, and I apologise for my use of "sharing". At the time I thought OBIS shared its processing results with GBIF. It doesn't. Publishers independently publish to OBIS and to GBIF, so the processing of scientificNameID happens in parallel. This comment deals with that.

@timrobertson100  generated a file that has taxonomic details for records with scientificNameID populated with WoRMS LSIDs. These are the ca 19000 cases in which there would be a disagreement in names if GBIF looked up the scientificNameID in WoRMS and compared it to the provided (original) scientificName.

The file contains numerous instances in which well-formed, formal taxonomic names have more than one scientificNameID. Here's one example. It's "Coscinoderma matthewsi", a misspelling of Coscinoderma mathewsi (Lendenfeld, 1886), urn:lsid:marinespecies.org:taxname:165090.

No. of records | scientificName | scientificNameID | WoRMS name
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165091 | Coscinoderma pesleonis (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165092 | Coscinoderma sinuosum (unaccepted synonym of Coscinoderma lamarcki)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165093 | Hippospongia ammata (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165094 | Hippospongia anfractuosa (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165095 | Hippospongia canaliculata (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165096 | Hippospongia cerebrum (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165097 | Hippospongia cylindrica (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165098 | Hippospongia decidua (unaccepted synonym of Hyattella sinuosa)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165099 | Hippospongia densa (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165100 | Hippospongia derasa (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165101 | Hippospongia equina (unaccepted synonym of Hippospongia communis)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165102 | Hippospongia equina var. meandriniformis (unaccepted synonym of Spongia (Spongia) barbara)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165103 | Hippospongia equina var. micropora (unaccepted synonym of Hippospongia micropora)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165104 | Hippospongia fistulosa (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165105 | Hippospongia flabellum (unaccepted synonym of Hippospongia canaliculata)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165106 | Hippospongia galea (accepted)
1 | Coscinoderma matthewsi | urn:lsid:marinespecies.org:taxname:165107 | Hippospongia gossypina (accepted)

This looks to me like an [incremental fill-down error](https://www.datafix.com.au/BASHing/2021-05-26.html) in a spreadsheet. The data compiler entered the correct LSID ("...165090"), then filled down incrementally over the next 17 records.

Because @timrobertson100  did not include a record ID I can't track the records back to GBIF or OBIS. However, the [OBIS page](https://obis.org/taxon/165090) for this sponge species says there were no invalid scientificNameIDs and no dropped records. Can you explain what is likely to have happened when the above records with incorrect scientificNameIDs were processed by OBIS? Did OBIS process each record according to its scientificNameID, thus incorrectly assigning the record to the wrong species?

Note that GBIF processed the records according to scientificName and correctly assigned all 17 records to Coscinoderma mathewsi (Lendenfeld, 1886).

**Comment by Mesibov** [comment]
@pieterprovoost, I found the dataset and the "Coscinoderma matthewsi" occurrences in GBIF. The dataset is [Vulnerable marine ecosystems in the South Pacific Ocean region](https://www.gbif.org/dataset/be62fffc-d0e7-45f2-9404-d4bc4322bc57) and was published by New Zealand's NIWA. Example occurrence [here](https://www.gbif.org/occurrence/1890314210). The [OBIS link](https://nzobisipt.niwa.co.nz/archive.do?r=vme_inverts).

**Comment by Mesibov** [comment]
@pieterprovoost, I've answered my own question: OBIS processed "Coscinoderma matthewsi" as Coscinoderma pesleonis based on the incorrect scientificNameID "...165091". The OBIS record id is d47e860b-eff5-42e2-9534-6d24a4810767 from the dataset 6c813a6c-86f7-4d45-beb5-33eebd8de938. How will OBIS correct existing errors of this kind and how will they be treated in future?

**Comment by pieterprovoost** [comment]
@Mesibov I'll have to run the check and report back to our data providers to get this fixed. I'll also add the flags in our QC procedures to prevent this in the future.

**Comment by timrobertson100** [comment]
Thanks to all for guidance on this. 

GBIF.org now processes `scientificNameID`, `taxonID` and `taxonConceptID` for configured identifier schemes.
The following flags have been added to help publishers and consumers understand how the identifiers have been used and where ambiguities may be detected.


| **Published**                                                                                                            | **Interpreted**                                                                                                                                                                                                                                                                                                                                                                                                                                      |
|--------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `TAXON_MATCH_SCIENTIFIC_NAME_ID_IGNORED`<br/> `TAXON_MATCH_TAXON_CONCEPT_ID_IGNORED`<br/> `TAXON_MATCH_TAXON_ID_IGNORED` | The …ID was not used when mapping the record to the GBIF backbone. This may indicate one of: <ul> <li>The ID uses a pattern not configured for use by GBIF</li> <li>The ID did not uniquely(!) identify a concept in the checklist</li> <li>The ID found a concept in the checklist that did not map to the backbone</li> <li>A different ID was used, or the record names were used, as no ID lookup successfully linked to the backbone</li> </ul> |
| `SCIENTIFIC_NAME_ID_NOT_FOUND`<br/> `TAXON_CONCEPT_ID_NOT_FOUND`<br/> `TAXON_ID_NOT_FOUND`                               | The …ID matched a known pattern, but it was not found in the associated checklist. The backbone lookup was performed using either the names or a different ID field from the record. This may indicate a poorly formatted identifier or may be caused by a newly created ID that isn't yet known in the version of the published checklist.                                                                                                          |
| `SCIENTIFIC_NAME_AND_ID_INCONSISTENT`                                                                                    | The scientificName provided in the occurrence record does not precisely match the name in the registered checklist when using the scientificNameID, taxonID or taxonConceptID to look it up. Publishers are advised to check the IDs are correct, or update the formatting of the names on their records.                                                                                                                                            |
| `TAXON_MATCH_NAME_AND_ID_AMBIGUOUS`                                                                                      | The GBIF Backbone concept was found using the scientificNameID, taxonID or taxonConceptID, but it differs from what would have been found if the classification names on the record were used. This may indicate a gap in the GBIF backbone, a poor mapping between the checklist and the backbone, or a mismatch between the classification names and the declared IDs (scientificNameID or taxonConceptID) on the occurrence record itself.        |

The GBIF.org site and API allow search using issues, such as [this example](https://www.gbif.org/occurrence/search?issue=SCIENTIFIC_NAME_AND_ID_INCONSISTENT&occurrence_status=present).

Initially the WoRMS LSIDs have been enabled to bring consistency to GBIF and OBIS processing and to work through any teething issues.
Other candidates for future use could be the [International Plant Name Index LSIDs](https://www.ipni.org/), [Swedish Dyntaxa LSIDs](https://doi.org/10.15468/j43wfc), [Catalogue of Life Identifiers](https://www.catalogueoflife.org/2022/03/23/name-ids), [Zoobank](https://zoobank.org/) 
LSIDs and [Index Fungorum](https://www.indexfungorum.org/names/IndexFungorumLSIDs.htm) LSIDs.

Because this issue has become very long, and the original request from @bart-v and OBIS team is implemented, this seems like a suitable point to try and close this issue. Please do open issues for specific improvement requests, bugs etc or in https://discourse.gbif.org/

**Comment by bart-v** [comment]
Excellent progress & impressive work.
Much appreciated
Thanks a lot @timrobertson100
